earth_enterprise/src/fusion/autoingest/sysman/khResourceManager.cpp

```diff
@@ -243,9 +243,13 @@ khResourceManager::InsertProvider(khResourceProviderProxy *proxy)
   // instantiate the Volumes that this provider will manage
   std::vector<std::string> volnames;
   GetHostVolumes(host, volnames);
-  for (std::vector<std::string>::const_iterator vn = volnames.begin();
-       vn != volnames.end(); ++vn) {
-    volumes[*vn] = new Volume(*vn, proxy);
+  for (const auto& vn : volnames) {
+        if (volumes.find(vn) != volumes.end())
+        {
+            delete volumes[vn];
```

Comment:
If we're going to include this change in this PR we need a better understanding of what it does. Why do duplicate volumes get added? Is there any difference between the new and old versions of the volumes? I think @tst-jkent is already working on this.

Reply:
I'm still running some tests to make sure, but so far this only gets called when gesystemmanager is started and when systemmanager is stopped the volumes are removed as expected. Still looking into whether or not there are any differences between the volumes each time the volumes are added.